web/src/app/pages/edit-survey/edit-survey.component.html

```diff
@@ -17,7 +17,9 @@
 <div class="page">
   <ground-header></ground-header>
 
-  <div class="container">
+  <!-- TODO(daoyu): replace it with a spinner component -->
```

Comment:
Instead of doing this in every page, can we create a component that wraps pages which require an active survey? The spinner and branching logic could be handled there. It could replace the top `div` and `ground-header` as well, eg

```
<ground-active-survey>
  <mat-nav-list class="left-menu">
  ...
```

If you don't intend to implement the spinner in this PR, please add a bug ref to the TODOs.

Reply:
AFAIK, Angular does not allow you to put Dom inside a component. You'd have to use `ng-template` like
```
<ground-active-survey [template]="temp" >
</ground-active-survey>

<ng-template #temp>
  <mat-nav-list class="left-menu">
  ...
</ng-template>
```
My past experience with `ng-template` is terrible(lots of limitations). And there aren't going to be many pages, I feel repeating it is much easier than creating a component and dealing with `ng-template`, so I chose this way.

We could extract this out to a component later if we start to have too many repetitions and feel confident with `ng-template`.

Reply:
Content projection is ALL YOU NEED

Reply:
Created bug #1170.